### PR TITLE
Density plot

### PR DESCRIPTION
**Pull request by aloctavodia**
This change the way the density is estimated following discussion in #403 and [discourse](https://discourse.pymc.io/t/bug-in-sample-prior-predictive/2210/6).

lala

### CODE REVIEW

File: arviz/plots/densityplot.py [hunk 1]
```diff
@@ -206,23 +210,25 @@ def _d_helper(
     ax : matplotlib axes
     """
     if vec.dtype.kind == "f":
-        density, lower, upper = _fast_kde(vec, bw=bw)
-        x = np.linspace(lower, upper, len(density))
-        hpd_ = hpd(vec, credible_interval)
-        cut = (x >= hpd_[0]) & (x <= hpd_[1])
+        if credible_interval < 0.999:
```
Comment (canyon289) [inline arviz/plots/densityplot.py]: Could the check be `credible_interval != 1` or is it important that it truly is less than `.999`?

Comment (aloctavodia) [reply arviz/plots/densityplot.py]: This seem a better choice.

**Comment by canyon289** [comment]
Just a question, I see the changes but I'm too dense to get what the change was before and after. If you get a second could you describe what was causing the issue and what the fix was? This is request is optional :)

**Comment by ahartikainen** [comment]
LGTM

**Pull request merged**